### ISSUE TITLE
Local Search Provider Hash Changes Every Build

### ISSUE THREAD

**Issue by wSedlacek** [vuejs/vitepress#4246]
### Describe the bug

When using the local search provider the hashes of the bundles will change every build.

### Reproduction

Set `search: { provider: 'local' }` then build, note hash of `@localSearchIndexroot`, then build again, see hash changed.

### Expected behavior

The `@localSearchIndexroot` hash should only change if the contents of the docs changed.

### System Info

```Text
sh: envinfo: command not found

Does this work on macOS?
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
> Does this work on macOS?

Yes it does. Looks like you ran it without `npx` in front. But anyway I can confirm the bug with latest version, no need for env info.

**Comment by wSedlacek** [comment]
> Yes it does. Looks like you ran it without npx in front. But anyway I can confirm the bug with latest version, no need for env info.

Hm, it worked when I retried. Seems like `npx` was behaving differently when I was using corepack and in a project directory, running it from the home directory resolved the issue.
